diff --git a/build-logic/build.gradle.kts b/build-logic/build.gradle.kts
index b9642d58..d1994d3f 100644
--- a/build-logic/build.gradle.kts
+++ b/build-logic/build.gradle.kts
@@ -8,6 +8,7 @@ plugins {
 }

 repositories {
+    /* %{DOKKA_IT_MAVEN_REPO}% */
     mavenCentral()
 }

diff --git a/build-logic/settings.gradle.kts b/build-logic/settings.gradle.kts
index 4b401d8b..fcfac2c6 100644
--- a/build-logic/settings.gradle.kts
+++ b/build-logic/settings.gradle.kts
@@ -11,6 +11,7 @@ dependencyResolutionManagement {
     versionCatalogs {
         create("libs") {
             from(files("../gradle/libs.versions.toml"))
+            version("dokka", providers.gradleProperty("dokka_it_dokka_version").get())
         }
     }
 }
diff --git a/build-logic/src/main/kotlin/kotlinx/io/conventions/kotlinx-io-dokka.gradle.kts b/build-logic/src/main/kotlin/kotlinx/io/conventions/kotlinx-io-dokka.gradle.kts
index eef54191..5a68581d 100644
--- a/build-logic/src/main/kotlin/kotlinx/io/conventions/kotlinx-io-dokka.gradle.kts
+++ b/build-logic/src/main/kotlin/kotlinx/io/conventions/kotlinx-io-dokka.gradle.kts
@@ -31,6 +31,9 @@ dokka {
     pluginsConfiguration.html {
         templatesDir = rootDir.resolve("dokka-templates")
     }
+    dokkaPublications.html {
+        failOnWarning.set(providers.gradleProperty("dokka_it_failOnWarning").map(String::toBoolean))
+    }
 }

 dependencies {
diff --git a/build.gradle.kts b/build.gradle.kts
index 0f855cb7..05bc4912 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -14,6 +14,7 @@ plugins {
 allprojects {
     properties["DeployVersion"]?.let { version = it }
     repositories {
+        /* %{DOKKA_IT_MAVEN_REPO}% */
         mavenCentral()
     }
 }
diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index d2028bc3..9de0d68e 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -2,7 +2,7 @@
 kotlin = "2.3.0"
 java = "8"
 multi-release-toolchain = "17"
-dokka = "2.2.0-Beta"
+dokka = "sanity-non-existent"
 kover = "0.9.3"
 benchmark = "0.4.14"
 jmh = "1.37"
diff --git a/settings.gradle.kts b/settings.gradle.kts
index 79515254..f4fb9730 100644
--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -7,6 +7,7 @@ pluginManagement {
     includeBuild("build-logic")

     repositories {
+        /* %{DOKKA_IT_MAVEN_REPO}% */
         mavenCentral()
         google()
         gradlePluginPortal()
