diff --git a/build.gradle b/build.gradle
index 73b566ae..d7c18353 100644
--- a/build.gradle
+++ b/build.gradle
@@ -57,14 +57,15 @@ buildscript {
     }

     repositories {
-        maven { url 'https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev' }
+        /* %{PROJECT_LOCAL_MAVEN_DIR}% */
+        //maven { url 'https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev' }
         // kotlin-dev with space redirector
         maven { url "https://cache-redirector.jetbrains.com/maven.pkg.jetbrains.space/kotlin/p/kotlin/dev" }
         mavenCentral()
         gradlePluginPortal()
         // For Dokka that depends on kotlinx-html
         maven { url "https://maven.pkg.jetbrains.space/public/p/kotlinx-html/maven" }
-        mavenLocal()
+        //mavenLocal()
     }

     configurations.classpath {
@@ -78,7 +79,7 @@ buildscript {
     dependencies {
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
         classpath "org.jetbrains.kotlin:kotlin-serialization:$kotlin_version"
-        classpath "org.jetbrains.dokka:dokka-gradle-plugin:$dokka_version"
+        classpath "org.jetbrains.dokka:dokka-gradle-plugin:${providers.gradleProperty("dokka_it_dokka_version").get()}"
         classpath "org.jetbrains.kotlinx:kover:$kover_version"
         classpath "org.jetbrains.kotlinx:binary-compatibility-validator:$validator_version"
         classpath "org.jetbrains.kotlinx:kotlinx-knit:$knit_version"
@@ -133,7 +134,8 @@ allprojects {
     if (build_snapshot_train) {
         // Snapshot-specific
         repositories {
-            mavenLocal()
+            /* %{PROJECT_LOCAL_MAVEN_DIR}% */
+            //mavenLocal()
             maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
         }
     }
@@ -154,14 +156,15 @@ allprojects {
     }

     repositories {
+        /* %{PROJECT_LOCAL_MAVEN_DIR}% */
         mavenCentral()
-        maven { url 'https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev' }
+        //maven { url 'https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev' }
         // kotlin-dev with space redirector
         maven { url "https://cache-redirector.jetbrains.com/maven.pkg.jetbrains.space/kotlin/p/kotlin/dev" }
         // For Dokka that depends on kotlinx-html
         maven { url "https://maven.pkg.jetbrains.space/public/p/kotlinx-html/maven" }
         // For local development
-        mavenLocal()
+        //mavenLocal()

     }
 }
diff --git a/buildSrc/build.gradle.kts b/buildSrc/build.gradle.kts
index c999bcd2..98afdae7 100644
--- a/buildSrc/build.gradle.kts
+++ b/buildSrc/build.gradle.kts
@@ -10,8 +10,9 @@ plugins {
 }

 repositories {
+    /* %{PROJECT_LOCAL_MAVEN_DIR}% */
     mavenCentral()
-    mavenLocal()
+    //mavenLocal()
     if (project.hasProperty("kotlin_repo_url")) {
         maven(project.properties["kotlin_repo_url"] as String)
     }
@@ -41,4 +42,3 @@ val kotlinVersion = run {
 dependencies {
     implementation(kotlin("gradle-plugin", kotlinVersion))
 }
-
diff --git a/integration-test/build.gradle b/integration-test/build.gradle
index dda68347..31a92d68 100644
--- a/integration-test/build.gradle
+++ b/integration-test/build.gradle
@@ -19,7 +19,8 @@ plugins {
 }

 repositories {
-    mavenLocal()
+    /* %{PROJECT_LOCAL_MAVEN_DIR}% */
+    //mavenLocal()
     mavenCentral()
     maven { url "https://cache-redirector.jetbrains.com/maven.pkg.jetbrains.space/kotlin/p/kotlin/dev" }
 }
diff --git a/integration-test/settings.gradle b/integration-test/settings.gradle
index f8cb2d87..beebf995 100644
--- a/integration-test/settings.gradle
+++ b/integration-test/settings.gradle
@@ -14,10 +14,11 @@ pluginManagement {
     }

     repositories {
+        /* %{PROJECT_LOCAL_MAVEN_DIR}% */
         mavenCentral()
         maven { url 'https://plugins.gradle.org/m2/' }
         maven { url "https://cache-redirector.jetbrains.com/maven.pkg.jetbrains.space/kotlin/p/kotlin/dev" }
-        mavenLocal()
+        //mavenLocal()
     }
 }
