diff --git a/build.gradle.kts b/build.gradle.kts
index 538b1804..923d0c86 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -52,6 +52,7 @@ allprojects {
         }
     }
     repositories {
+        /* %{DOKKA_IT_MAVEN_REPO}% */
         mavenCentral()
         maven("https://redirector.kotlinlang.org/maven/dev")
     }
diff --git a/buildSrc/build.gradle.kts b/buildSrc/build.gradle.kts
index 5c73ad8b..a6899605 100644
--- a/buildSrc/build.gradle.kts
+++ b/buildSrc/build.gradle.kts
@@ -26,6 +26,7 @@ repositories {
         maven("https://oss.sonatype.org/content/repositories/snapshots")
     }

+    /* %{DOKKA_IT_MAVEN_REPO}% */
     mavenCentral()
     maven("https://redirector.kotlinlang.org/maven/dev")
 }
diff --git a/buildSrc/settings.gradle.kts b/buildSrc/settings.gradle.kts
index b450fd5e..9f189439 100644
--- a/buildSrc/settings.gradle.kts
+++ b/buildSrc/settings.gradle.kts
@@ -14,6 +14,7 @@ dependencyResolutionManagement {
                 logger.info("Overriding Kotlin version in buildSrc: $overriddenVersion")
                 version("kotlin", overriddenVersion)
             }
+            version("dokka", providers.gradleProperty("dokka_it_dokka_version").get())
         }
     }
 }
diff --git a/buildSrc/src/main/kotlin/dokka-conventions.gradle.kts b/buildSrc/src/main/kotlin/dokka-conventions.gradle.kts
index d64d46a9..74f00e7a 100644
--- a/buildSrc/src/main/kotlin/dokka-conventions.gradle.kts
+++ b/buildSrc/src/main/kotlin/dokka-conventions.gradle.kts
@@ -16,9 +16,11 @@ dokka {
         templatesDir = rootDir.resolve("dokka-templates")
     }

+    dokkaPublications.html {
+        failOnWarning.set(providers.gradleProperty("dokka_it_failOnWarning").map(String::toBoolean))
+    }
     dokkaSourceSets.configureEach {
         includes.from(rootDir.resolve("dokka/moduledoc.md").path)
-        reportUndocumented = true
         skipDeprecated = true

         perPackageOption {
diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index 126e9fda..85e8aae7 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -1,7 +1,7 @@
 [versions]
 kotlin = "2.3.0"
 kover = "0.8.2"
-dokka = "2.2.0-Beta"
+dokka = "sanity-non-existent"
 knit = "0.5.0"
 bcv = "0.18.1"
 animalsniffer = "1.7.1"
diff --git a/settings.gradle.kts b/settings.gradle.kts
index 64d86bb4..df2f48c3 100644
--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -88,6 +88,7 @@ dependencyResolutionManagement {
                 logger.info("Overriding Kotlin version: $overriddenVersion")
                 version("kotlin", overriddenVersion)
             }
+            version("dokka", providers.gradleProperty("dokka_it_dokka_version").get())
         }
     }
 }
