diff --git a/build.gradle b/build.gradle
index 73b566ae..c7224a7a 100644
--- a/build.gradle
+++ b/build.gradle
@@ -76,9 +76,10 @@ buildscript {
     }
 
     dependencies {
+        def dokkaVersion = System.getenv('DOKKA_VERSION')
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
         classpath "org.jetbrains.kotlin:kotlin-serialization:$kotlin_version"
-        classpath "org.jetbrains.dokka:dokka-gradle-plugin:$dokka_version"
+        classpath "org.jetbrains.dokka:dokka-gradle-plugin:$dokkaVersion"
         classpath "org.jetbrains.kotlinx:kover:$kover_version"
         classpath "org.jetbrains.kotlinx:binary-compatibility-validator:$validator_version"
         classpath "org.jetbrains.kotlinx:kotlinx-knit:$knit_version"
diff --git a/gradle.properties b/gradle.properties
index 5ef66cd9..7e7323fd 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -13,7 +13,7 @@ kotlin.version.snapshot=1.9.255-SNAPSHOT
 
 junit_version=4.12
 jackson_version=2.10.0.pr1
-dokka_version=1.8.10
+dokka_version=non-existing-sanity-check-SNAPSHOT
 native.deploy=
 validator_version=0.13.2
 knit_version=0.5.0-Beta
