diff --git a/build.gradle b/build.gradle
index e7d405e12..db5dcec66 100644
--- a/build.gradle
+++ b/build.gradle
@@ -52,17 +52,27 @@ buildscript {
     }

     repositories {
+        /* %{PROJECT_LOCAL_MAVEN_DIR}% */
         mavenCentral()
         maven { url "https://plugins.gradle.org/m2/" }
         CommunityProjectsBuild.addDevRepositoryIfEnabled(delegate, project)
-        mavenLocal()
+        //mavenLocal()
     }

     dependencies {
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-        classpath "org.jetbrains.dokka:dokka-gradle-plugin:$dokka_version"
+        classpath "org.jetbrains.dokka:dokka-gradle-plugin:${providers.gradleProperty("dokka_it_dokka_version").get()}"
         classpath "org.jetbrains.kotlinx:atomicfu-gradle-plugin:$atomicfu_version"
-        classpath "org.jetbrains.kotlinx:kotlinx-knit:$knit_version"
+        classpath("org.jetbrains.kotlinx:kotlinx-knit:$knit_version") {
+            exclude(group: "org.jetbrains.kotlinx", module: "dokka-pathsaver-plugin")
+        }
+        classpath("org.jetbrains.kotlinx:dokka-pathsaver-plugin:$knit_version") {
+            exclude(group: "org.jetbrains.dokka", module: "templating-plugin")
+            exclude(group: "org.jetbrains.dokka", module: "dokka-base")
+        }
+        classpath("org.jetbrains.dokka:templating-plugin:${providers.gradleProperty("dokka_it_dokka_version").get()}")
+        classpath("org.jetbrains.dokka:dokka-base:${providers.gradleProperty("dokka_it_dokka_version").get()}")
+
         classpath "com.github.node-gradle:gradle-node-plugin:$gradle_node_version"
         classpath "org.jetbrains.kotlinx:binary-compatibility-validator:$binary_compatibility_validator_version"
         classpath "ru.vyarus:gradle-animalsniffer-plugin:1.5.4" // Android API check
@@ -102,11 +112,11 @@ allprojects {
         kotlin_version = rootProject.properties['kotlin_snapshot_version']
     }

-    if (using_snapshot_version) {
-        repositories {
-            mavenLocal()
-        }
-    }
+    //if (using_snapshot_version) {
+    //    repositories {
+    //        mavenLocal()
+    //    }
+    //}

     ext.unpublished = unpublished

@@ -135,6 +145,7 @@ apiValidation {
 // Configure repositories
 allprojects {
     repositories {
+        /* %{PROJECT_LOCAL_MAVEN_DIR}% */
         /*
          * google should be first in the repository list because some of the play services
          * transitive dependencies was removed from jcenter, thus breaking gradle dependency resolution
@@ -323,7 +334,12 @@ knit {
 knitPrepare.dependsOn getTasksByName("dokkaHtmlMultiModule", true)

 dependencies {
-    dokkaHtmlMultiModulePlugin("org.jetbrains.kotlinx:dokka-pathsaver-plugin:$knit_version")
+    dokkaHtmlMultiModulePlugin("org.jetbrains.kotlinx:dokka-pathsaver-plugin:$knit_version") {
+        exclude(group: "org.jetbrains.dokka", module: "templating-plugin")
+        exclude(group: "org.jetbrains.dokka", module: "dokka-base")
+    }
+    dokkaHtmlMultiModulePlugin("org.jetbrains.dokka:templating-plugin:${providers.gradleProperty("dokka_it_dokka_version").get()}")
+    dokkaHtmlMultiModulePlugin("org.jetbrains.dokka:dokka-base:${providers.gradleProperty("dokka_it_dokka_version").get()}")
 }

 // Opt-in for build scan in order to troubleshoot Gradle on TC
diff --git a/buildSrc/build.gradle.kts b/buildSrc/build.gradle.kts
index ae54ad0f6..00963f5b2 100644
--- a/buildSrc/build.gradle.kts
+++ b/buildSrc/build.gradle.kts
@@ -13,6 +13,7 @@ val buildSnapshotTrain = properties["build_snapshot_train"]?.toString()?.toBoole
 val kotlinDevUrl = project.rootProject.properties["kotlin_repo_url"] as? String

 repositories {
+    /* %{PROJECT_LOCAL_MAVEN_DIR}% */
     mavenCentral()
     if (cacheRedirectorEnabled) {
         maven("https://cache-redirector.jetbrains.com/plugins.gradle.org/m2")
@@ -22,9 +23,9 @@ repositories {
     if (!kotlinDevUrl.isNullOrEmpty()) {
         maven(kotlinDevUrl)
     }
-    if (buildSnapshotTrain) {
-        mavenLocal()
-    }
+    //if (buildSnapshotTrain) {
+    //    mavenLocal()
+    //}
 }

 val gradleProperties = Properties().apply {
@@ -49,12 +50,12 @@ dependencies {
      * our version of Gradle bundles Kotlin 1.4.x and can read metadata only up to 1.5.x,
      * thus we're excluding stdlib compiled with 1.6.0 from dependencies.
      */
-    implementation("org.jetbrains.dokka:dokka-gradle-plugin:${version("dokka")}") {
+    implementation("org.jetbrains.dokka:dokka-gradle-plugin:${providers.gradleProperty("dokka_it_dokka_version").get()}") {
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib-jdk8")
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib-jdk7")
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib")
     }
-    implementation("org.jetbrains.dokka:dokka-core:${version("dokka")}") {
+    implementation("org.jetbrains.dokka:dokka-core:${providers.gradleProperty("dokka_it_dokka_version").get()}") {
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib-jdk8")
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib-jdk7")
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib")
diff --git a/buildSrc/settings.gradle.kts b/buildSrc/settings.gradle.kts
index c2e859f65..9cc749a1f 100644
--- a/buildSrc/settings.gradle.kts
+++ b/buildSrc/settings.gradle.kts
@@ -4,6 +4,7 @@
 pluginManagement {
     val build_snapshot_train: String? by settings
     repositories {
+        /* %{PROJECT_LOCAL_MAVEN_DIR}% */
         val cacheRedirectorEnabled = System.getenv("CACHE_REDIRECTOR")?.toBoolean() == true
         if (cacheRedirectorEnabled) {
             println("Redirecting repositories for buildSrc buildscript")
@@ -11,8 +12,8 @@ pluginManagement {
         } else {
             maven("https://plugins.gradle.org/m2")
         }
-        if (build_snapshot_train?.toBoolean() == true) {
-            mavenLocal()
-        }
+        //if (build_snapshot_train?.toBoolean() == true) {
+        //    mavenLocal()
+        //}
     }
 }
diff --git a/gradle.properties b/gradle.properties
index 3d9431be0..9af01ef86 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -14,7 +14,7 @@ atomicfu_version=0.21.0
 knit_version=0.5.0-Beta
 html_version=0.7.2
 lincheck_version=2.18.1
-dokka_version=1.8.10
+dokka_version=non-existing-sanity-check-SNAPSHOT
 byte_buddy_version=1.10.9
 reactor_version=3.4.1
 reactive_streams_version=1.0.3
diff --git a/gradle/dokka.gradle.kts b/gradle/dokka.gradle.kts
index ba6956aa8..4a5d27e1a 100644
--- a/gradle/dokka.gradle.kts
+++ b/gradle/dokka.gradle.kts
@@ -21,7 +21,13 @@ fun GradleDokkaSourceSetBuilder.makeLinkMapping(projectDir: File) {
 val knit_version: String by project
 tasks.withType(DokkaTaskPartial::class).configureEach {
     dependencies {
-        plugins("org.jetbrains.kotlinx:dokka-pathsaver-plugin:$knit_version")
+        val knit_version: String by project
+        plugins("org.jetbrains.kotlinx:dokka-pathsaver-plugin:$knit_version") {
+            exclude(group = "org.jetbrains.dokka", module = "templating-plugin")
+            exclude(group = "org.jetbrains.dokka", module = "dokka-base")
+        }
+        plugins("org.jetbrains.dokka:templating-plugin:${providers.gradleProperty("dokka_it_dokka_version").get()}")
+        plugins("org.jetbrains.dokka:dokka-base:${providers.gradleProperty("dokka_it_dokka_version").get()}")
     }
 }

diff --git a/settings.gradle b/settings.gradle
index 151c087fd..e4433c24f 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -9,7 +9,8 @@ pluginManagement {
     }

     repositories {
-        maven { url "https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev/" }
+        /* %{PROJECT_LOCAL_MAVEN_DIR}% */
+        //maven { url "https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev/" }
         gradlePluginPortal()
     }
 }
