diff --git a/build.gradle.kts b/build.gradle.kts
index b6ea33000..0c644c14e 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -10,6 +10,7 @@ buildscript {
     }

     repositories {
+        /* %{DOKKA_IT_MAVEN_REPO}% */
         mavenCentral()
         maven(url = "https://plugins.gradle.org/m2/")
         addDevRepositoryIfEnabled(this, project)
@@ -21,8 +22,10 @@ buildscript {
         // The corresponding issue in kotlinx-atomicfu: https://github.com/Kotlin/kotlinx-atomicfu/issues/384
         classpath("org.jetbrains.kotlinx:atomicfu-gradle-plugin:${version("atomicfu")}")
         classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${version("kotlin")}")
-        classpath("org.jetbrains.dokka:dokka-gradle-plugin:${version("dokka")}")
-        classpath("org.jetbrains.kotlinx:kotlinx-knit:${version("knit")}")
+        classpath("org.jetbrains.dokka:dokka-gradle-plugin:${providers.gradleProperty("dokka_it_dokka_version").get()}")
+        classpath("org.jetbrains.kotlinx:kotlinx-knit:${version("knit")}") {
+            exclude("org.jetbrains.kotlinx", "dokka-pathsaver-plugin")
+        }
         classpath("ru.vyarus:gradle-animalsniffer-plugin:${version("animalsniffer")}") // Android API check
         classpath("org.jetbrains.kotlin:atomicfu:${version("kotlin")}")
         classpath("org.jetbrains.kotlinx:kover-gradle-plugin:${version("kover")}")
@@ -67,6 +70,7 @@ apply(plugin = "kover-conventions")
 // Configure repositories
 allprojects {
     repositories {
+        /* %{DOKKA_IT_MAVEN_REPO}% */
         /*
          * google should be first in the repository list because some of the play services
          * transitive dependencies was removed from jcenter, thus breaking gradle dependency resolution
diff --git a/buildSrc/build.gradle.kts b/buildSrc/build.gradle.kts
index ef767acf3..c5c7a80bc 100644
--- a/buildSrc/build.gradle.kts
+++ b/buildSrc/build.gradle.kts
@@ -9,6 +9,7 @@ val buildSnapshotTrain = properties["build_snapshot_train"]?.toString()?.toBoole
 val kotlinDevUrl = project.providers.gradleProperty("kotlin_repo_url").orNull

 repositories {
+    /* %{DOKKA_IT_MAVEN_REPO}% */
     mavenCentral()
     if (cacheRedirectorEnabled) {
         maven("https://cache-redirector.jetbrains.com/plugins.gradle.org/m2")
@@ -51,12 +52,12 @@ dependencies {
      * our version of Gradle bundles Kotlin 1.4.x and can read metadata only up to 1.5.x,
      * thus we're excluding stdlib compiled with 1.6.0 from dependencies.
      */
-    implementation("org.jetbrains.dokka:dokka-gradle-plugin:${version("dokka")}") {
+    implementation("org.jetbrains.dokka:dokka-gradle-plugin:${providers.gradleProperty("dokka_it_dokka_version").get()}") {
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib-jdk8")
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib-jdk7")
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib")
     }
-    implementation("org.jetbrains.dokka:dokka-core:${version("dokka")}") {
+    implementation("org.jetbrains.dokka:dokka-core:${providers.gradleProperty("dokka_it_dokka_version").get()}") {
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib-jdk8")
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib-jdk7")
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib")
@@ -69,6 +70,8 @@ dependencies {
         exclude(group = "org.jetbrains.kotlin", module = "kotlin-stdlib")
     }
     implementation("org.jetbrains.kotlinx:kotlinx-benchmark-plugin:${version("benchmarks")}")
-    implementation("org.jetbrains.kotlinx:kotlinx-knit:${version("knit")}")
+    implementation("org.jetbrains.kotlinx:kotlinx-knit:${version("knit")}") {
+        exclude("org.jetbrains.kotlinx", "dokka-pathsaver-plugin")
+    }
     implementation("org.jetbrains.kotlinx:atomicfu-gradle-plugin:${version("atomicfu")}")
 }
diff --git a/buildSrc/src/main/kotlin/dokka-conventions.gradle.kts b/buildSrc/src/main/kotlin/dokka-conventions.gradle.kts
index d669d4b37..10c177bc8 100644
--- a/buildSrc/src/main/kotlin/dokka-conventions.gradle.kts
+++ b/buildSrc/src/main/kotlin/dokka-conventions.gradle.kts
@@ -41,6 +41,7 @@ private fun Project.condigureDokkaSetup() {
     dokka {
         dokkaPublications.configureEach {
             suppressInheritedMembers = true
+            failOnWarning.set(providers.gradleProperty("dokka_it_failOnWarning").map(String::toBoolean))
         }
         dokkaSourceSets.configureEach {
             jdkVersion = 11
diff --git a/gradle.properties b/gradle.properties
index d9915bdaa..5fbc12590 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -14,7 +14,7 @@ junit_version=4.12
 junit5_version=5.7.0
 knit_version=0.5.0
 lincheck_version=2.18.1
-dokka_version=2.2.0-Beta
+dokka_version=sanity-non-existent
 byte_buddy_version=1.17.6
 reactor_version=3.4.1
 reactor_docs_version=3.4.5
diff --git a/settings.gradle.kts b/settings.gradle.kts
index 423c613f3..c51ec4237 100644
--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -6,6 +6,7 @@ pluginManagement {
     }

     repositories {
+        /* %{DOKKA_IT_MAVEN_REPO}% */
         maven(url = "https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev/")
         gradlePluginPortal()
     }
