diff --git a/build.gradle b/build.gradle
index 934d4c220..38321990e 100644
--- a/build.gradle
+++ b/build.gradle
@@ -131,6 +131,7 @@ allprojects {
         google()
         mavenCentral()
         CommunityProjectsBuild.addDevRepositoryIfEnabled(delegate, project)
+        mavenLocal()
     }
 }
 
diff --git a/buildSrc/build.gradle.kts b/buildSrc/build.gradle.kts
index 785d13fdb..7fb19f467 100644
--- a/buildSrc/build.gradle.kts
+++ b/buildSrc/build.gradle.kts
@@ -25,6 +25,7 @@ repositories {
     if (buildSnapshotTrain) {
         mavenLocal()
     }
+    mavenLocal()
 }
 
 kotlinDslPluginOptions {
diff --git a/buildSrc/settings.gradle.kts b/buildSrc/settings.gradle.kts
index c2e859f65..43dc4f749 100644
--- a/buildSrc/settings.gradle.kts
+++ b/buildSrc/settings.gradle.kts
@@ -14,5 +14,6 @@ pluginManagement {
         if (build_snapshot_train?.toBoolean() == true) {
             mavenLocal()
         }
+        mavenLocal()
     }
 }
diff --git a/gradle.properties b/gradle.properties
index db7cf099b..9c13c46fd 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -14,7 +14,7 @@ atomicfu_version=0.18.5
 knit_version=0.4.0
 html_version=0.7.2
 lincheck_version=2.14.1
-dokka_version=1.7.20
+dokka_version=for-integration-tests-SNAPSHOT
 byte_buddy_version=1.10.9
 reactor_version=3.4.1
 reactive_streams_version=1.0.3
diff --git a/settings.gradle b/settings.gradle
index 151c087fd..e578bdb93 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -11,6 +11,7 @@ pluginManagement {
     repositories {
         maven { url "https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev/" }
         gradlePluginPortal()
+        mavenLocal()
     }
 }

